File: src/main/scala/mmlTests/a201701b/java/gpt4/EventImpl.java
package mmlTests.a201701b.java.gpt4;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Iterator;
import java.util.List;
import java.util.Optional;
import java.util.function.Supplier;

import mmlTests.a201701b.java.Event;
import mmlTests.a201701b.java.Trace;
import mmlTests.a201701b.java.TraceFactory;

public class EventImpl<X> implements Event<X> {

	private final int time;
	private final X value;

	public EventImpl(int time, X value) {
		this.time = time;
		this.value = value;
	}

	@Override
	public int getTime() {
		return time;
	}

	@Override
	public X getValue() {
		return value;
	}

}

File: src/main/scala/mmlTests/a201701b/scala/gpt4/EventImpl.scala
package mmlTests.a201701b.scala.gpt4

import java.util.function.Supplier
import java.util.{ArrayList, Iterator, Optional}
import mmlTests.a201701b.scala.Event
import mmlTests.a201701b.scala.Trace
import mmlTests.a201701b.scala.TraceFactory

class EventImpl[X](private val time: Int, private val value: X) extends Event[X] {
  override def getTime: Int = time
  override def getValue: X = value
}




